lib/IDE/SourceEntityWalker.cpp

```diff
@@ -607,10 +607,7 @@ ASTWalker::PreWalkResult<Expr *> SemaAnnotator::walkToExprPre(Expr *E) {
         return Action::Stop();
     }
 
-    if (auto rewritten = ME->getRewritten()) {
-      if (!rewritten->walk(*this))
-        return Action::Stop();
-    }
+    // Do not walk into ME->getRewritten() because it's not what the user wrote.
```

Comment:
I agree that this would be nice to have, but given nothing will handle that properly right now IMO dropping here makes sense to me.
  - No way to refer to the macro buffer at the moment (and *everything* has to handle whatever we end up doing here, ie. we don't want to try open it from the filesystem anywhere)
  - Rename has no concept of a reference it can't rename
  - Not quite sure what your last point means 🤔